examples/hosts/hosts-sample/README.md

```diff
@@ -1,39 +1,88 @@
 # @fluid-internal/hosts-sample
 
-The Fluid loader is all that is needed to load any Fluid document. This example walks through all the steps to
-create, initialize, and then make use of the loader. And does so in a literate programming like style to provide
-more detail on each line of the code.
+This example demonstrates how to host Fluid components in a web application.
+The Fluid loader is needed to load a Fluid document. The example walks through all the steps to
+create, initialize, and then make use of the loader in under 200 lines of code.
+And does so in a literate programming-like style to provide more detail on each line of the code.
 
-There are other packages which provide simple APIs to make use of the loader. For example @fluidframework/base-host.
-These may be better starting options when integrating the loader into your own code. But this example will show all
-the steps needed to create and use the Fluid loader. And it still comes in under 200 lines
-of code.
+<!-- AUTO-GENERATED-CONTENT:START (GET_STARTED:tinylicious=true) -->
+<!-- The getting started instructions are automatically generated.
+To update them, edit md-magic.config.js in the root of the repo, then run npm run readme:update -->
 
-## Build steps
+## Getting Started
 
-Getting up and running is simple.
+You can run this example using the following steps:
 
-```bash
-npm install
-npm start
+1. Run `npm install` and `npm run build:fast -- --nolint` from the `FluidFramework` root directory.
+   a. For an even faster build, you can add the package name to the build command, like this:
+   `npm run build:fast -- --nolint @fluid-internal/hosts-sample`
+1. In a separate terminal, start a Tinylicious server by following the instructions in [Tinylicious](../../../server/tinylicious).
+1. Run `npm run start` from this directory (examples/hosts/hosts-sample) and open <http://localhost:8080> in a web browser to see the app running.
+ <!-- AUTO-GENERATED-CONTENT:END -->
+
+## Using Example
+Upon navigating to the example application home URL with no parameters, it will create a new document with an auto-generated ID and load it using the default package name and version. When loaded correctly a placeholder component will be rendered in the content `div` element.
+
+![Home Page](./images/home-page.png)
+
+The application URL has the following format:
 ```
+http://localhost:8080/[#document-id]
+```
+, where `document-id` is an alphanumerical string representing the unique Fluid document ID.
+
+The application also provides the code details panel allowing to load different code versions and propose code upgrades.
+
+![Code Details](./images/code-details.png)
+
+Consider the following code upgrade scenario (see the image below):
+
+1. User A creates the initial document using the version 1.0.0.
+1. User A and User B load the document using a newer code version 2.0.0.
```

Comment:
This seems strange to me - I understand that a code loader is technically free to elect to load 2.0.0 here, but it seems like the more typical flow would be to load a 1.x package until 2.x is proposed?

Reply:
The sample covers the scenario when the app loads the latest package version available when opening the document. The opposite case simply doesn't work well with the current code upgrade mechanism.

Reply:
As I mentioned earlier
> having the application loading a newer code than in the quorum is fundamental for the code upgrade to work.